cmd/spire-server/cli/run/run.go

```diff
@@ -34,6 +35,7 @@ type runConfig struct {
 	Server        serverRunConfig            `hcl:"server"`
 	PluginConfigs catalog.HCLPluginConfigMap `hcl:"plugins"`
 	Telemetry     telemetry.FileConfig       `hcl:"telemetry"`
+	HealthChecks  health.Config              `hcl:"health_checks"`
```

Comment:
I am curious to hear your thoughts on how this might apply to spire agent in the future. I'm personally a little hesitant to allow the agent to bind to a non-localhost address.

If we think that there might be differences in what is made available to the server vs the agent, then this config might better suited to be put directly into the `server { ... }` and `agent { ... }` config blocks.

Lastly (as a general comment) we have been trying to keep configuration complexity to a minimum. The more that we can have sane defaults for, the better. The less we can expose that has the potential for misconfiguration or abuse, the better.

Reply:
Here's one possible example:
```
server {
    healthcheck_listener_enabled = true
    healthcheck_bind_address = "0.0.0.0:80"
}

...

agent {
    healthcheck_listener_enabled = true
}

Reply:
My use-case is fine with plaintext being localhost-only, and this will match the prometheus one (just set a port).

Reply:
Current config sample:

```health_checks {
       listener_enabled = true
       bind_address = "127.0.0.1"
       bind_port = "80"
       ready_path = "/ready"
       live_path = "/live"
}```

It's server-only, but we could disallow setting bind_address in the agent if you want (though similarly to the discussion in prometheus metrics, not sure that's worthwhile